packages/client/src/v2-events/features/workqueues/EventOverview/components/EventHistory/EventHistoryDialog/components/ActionTypeSpecificContent.tsx

```diff
@@ -9,36 +9,59 @@
  * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
  */
 import React from 'react'
+import { z } from 'zod'
 import {
-  ActionDocument,
   ActionType,
   DeclarationActions,
-  EventDocument
+  EventDocument,
+  PrintCertificateAction,
+  RequestedCorrectionAction
 } from '@opencrvs/commons/client'
+import {
+  DECLARATION_ACTION_UPDATE,
+  EventHistoryActionDocument
+} from '@client/v2-events/features/events/actions/correct/useActionForHistory'
 import { RequestCorrection } from './RequestCorrection'
 import { PrintCertificate } from './PrintCertificate'
 import { DeclarationUpdate } from './DeclarationUpdate'
 
+const DeclarationActionsWithUpdate = z.enum([
+  ...DeclarationActions.options,
+  DECLARATION_ACTION_UPDATE
+])
+
 export function ActionTypeSpecificContent({
   action,
   fullEvent
 }: {
-  action: ActionDocument
+  action: EventHistoryActionDocument
   fullEvent: EventDocument
 }) {
   const { type } = action
 
-  const isDeclarationAction = DeclarationActions.safeParse(type).success
+  const isDeclarationAction =
+    DeclarationActionsWithUpdate.safeParse(type).success
+
   if (isDeclarationAction) {
     return <DeclarationUpdate action={action} fullEvent={fullEvent} />
   }
 
   if (type === ActionType.REQUEST_CORRECTION) {
```

Comment:
using `action.type === ActionType.REQUEST_CORRECTION` should narrow down the type of action